merlin/models/data/advertising/criteo/dataset.py

```diff
@@ -0,0 +1,209 @@
+import glob
+import os
+from pathlib import Path
+from typing import Optional, Tuple, Union
+
+import numpy as np
+
+import merlin.io
+from merlin.core.utils import download_file
+from merlin.models.data import BASE_PATH
+from merlin.models.utils.example_utils import workflow_fit_transform
+from merlin.schema import Tags
+
+try:
+    import nvtabular as nvt
+
+    Workflow = nvt.Workflow
+except ImportError:
+    Workflow = None
+
+
+def get_criteo(
+    path: Union[str, Path] = None,
+    num_days: int = 2,
+    overwrite: bool = False,
+    transformed_name: str = "transformed",
+    nvt_workflow: Optional[Workflow] = None,
+    **kwargs,
+) -> Tuple[merlin.io.Dataset, merlin.io.Dataset]:
+    """
+    Get the Criteo 1TB Click Logs dataset, more info can be found
+    [here](https://ailab.criteo.com/download-criteo-1tb-click-logs-dataset/).
+
+    This dataset contains feature values and click feedback for millions of display
+    ads. Its purpose is to benchmark algorithms for clickthrough rate (CTR) prediction.
+    It is similar, but larger, to the dataset released for the Display Advertising
+    Challenge hosted by Kaggle: https://www.kaggle.com/c/criteo-display-ad-challenge
+
+    This function will download the data if not already present. Note, this can
+    In case you want to use this dataset to run our tutorials, you can also opt for synthetic data.
+    Synthetic data can be generated by running::
+        from merlin.models.data.synthetic import generate_data
+
+        data = generate_data("criteo")
+
+    Parameters
+    ----------
+    path: str or Path, optional
+        Path to the dataset. If not set will default to
+        the 'merlin-models-data` directory in your home folder
+    num_days: int, optional
+        Number of days to download, should be between 2 & 23. Default is 2.
+        The last day will be used for the valid set, the rest for train.
```

Comment:
We could. I did it like this since all papers that use the dataset do it like this. But there might be cases where you want to increase the size of the valid-set. Let me add an option for that.